src/main/java/de/unistuttgart/bugfinder/solution/bug/BugMapper.java

```diff
@@ -0,0 +1,52 @@
+package de.unistuttgart.bugfinder.solution.bug;
+
+import de.unistuttgart.bugfinder.code.word.WordMapper;
+import de.unistuttgart.bugfinder.code.word.WordRepository;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@FieldDefaults(level = AccessLevel.PUBLIC)
+public class BugMapper {
+
+  @Autowired
+  private WordMapper wordMapper;
+
+  @Autowired
+  private WordRepository wordRepository;
+
+  public BugDTO toDTO(final Bug bug) {
+    return new BugDTO(bug.getId(), bug.getWord().getId(), bug.getErrorType(), bug.getCorrectValue());
+  }
+
+  public List<BugDTO> toDTO(final Collection<Bug> bugs) {
+    return bugs.stream().map(this::toDTO).collect(Collectors.toList());
+  }
+
+  public Optional<BugDTO> toDTO(final Optional<Bug> bug) {
+    return bug.isEmpty() ? Optional.empty() : Optional.of(toDTO(bug.get()));
+  }
+
+  public Bug fromDTO(final BugDTO bugDTO) {
+    return new Bug(
+      bugDTO.getId(),
+      wordRepository.findById(bugDTO.getWordId()).orElseThrow(),
```

Comment:
```suggestion
      wordRepository.findById(bugDTO.getWordId()).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("word %s does not exist", bugDTO.getWordId()))),
```
Or do you have a better idea than a `404` here? It is somewhat fitting, but is there something better?

Reply:
Still relevant.